addons/test_mail_full/tests/test_portal.py

```diff
@@ -295,45 +279,53 @@ def test_customer_access_logged_without_access(self):
         self.authenticate(self.env.user.login, self.env.user.login)
         res = self.url_open(self.record_access_url)
         self.assertEqual(res.status_code, 200)
-        self.assert_URL(res.url, f'/my/test_portal/{self.record_portal.id}')
+        parsed_url = url_parse(res.url)
+        self.assertEqual(parsed_url.path, f'/my/test_portal/{self.record_portal.id}')
 
     @users('portal_test')
     def test_customer_access_logged_without_access_wrong_token(self):
         """Check that it redirects to discuss when logged customer has no access to the record and token is invalid."""
         self.authenticate(self.env.user.login, self.env.user.login)
         res = self.url_open(self.record_access_url_wrong_token)
         self.assertEqual(res.status_code, 200)
-        self.assert_URL(res.url, '/my', {'action': 'mail.action_discuss'})
+        parsed_url = url_parse(res.url)
+        self.assertEqual(parsed_url.path, '/my')
+        self.assertEqual(parsed_url.query, 'subpath=action-mail.action_discuss')
 
     def test_customer_access_not_logged(self):
         """Check that the access link redirects the customer (not logged) to the portal for viewing the record."""
         res = self.url_open(self.record_access_url)
         self.assertEqual(res.status_code, 200)
         self.assertIn(f'/my/test_portal/{self.record_portal.id}', res.url)
-        self.assert_URL(res.url, f'/my/test_portal/{self.record_portal.id}')
+        parsed_url = url_parse(res.url)
+        self.assertEqual(parsed_url.path, f'/my/test_portal/{self.record_portal.id}')
 
     def test_customer_access_not_logged_wrong_token(self):
         """Check that the access link redirect the customer to login when the token is invalid."""
         res = self.url_open(self.record_access_url_wrong_token)
         self.assertEqual(res.status_code, 200)
-        self.assert_URL(res.url, '/web/login', {'model': 'mail.test.portal', 'id': str(self.record_portal.id)},
-                        expected_query='redirect=')
+        parsed_url = url_parse(res.url)
+        self.assertEqual(parsed_url.path, '/web/login')
+        self.assertIn(f'model=mail.test.portal&id={self.record_portal.id}', parsed_url.fragment)
+        self.assertEqual(parsed_url.query, 'redirect=')
```

Comment:
This is also not a webclient url, it's the login, for me it should stay as it is.

Reply:
I've made this changes as i am not using `assert_URL` anymore.